include/realtime_tools/thread_priority.hpp

```diff
@@ -46,6 +46,18 @@ bool has_realtime_kernel();
  */
 bool configure_sched_fifo(int priority);
 
+/**
+ * Configure the caller thread affinity - Tell the scheduler to prefer a certain core for the current thread
+ * \param[in] core - the number of the core
+ * \returns true if configuring the scheduler succeeded
+*/
+bool set_preferred_core(const int core = -1);
```

Comment:
```suggestion
/**
 * Configure the caller thread affinity - Tell the scheduler to prefer a certain core for the current thread
 * \param[in] core - the number of the core
 * When parsed -1, the CPU affinity is reset to all CPUs
 * \returns true if configuring the scheduler succeeded
*/
bool set_cpu_affinity(const int core);
```

I would rather make the naming more standard as known to everyone